### DISCUSSION TITLE
How get the Results from Add-On Steel-Design and Stress-Strain-Analysis

### DISCUSSION THREAD

**Discussion by tobiasneulichedl** [Dlubal-Software/RFEM_Python_Client#360] [Q&A]
Hi!
I would like to get Results from the two Add-Ons Steel-Design and Stress-Strain-Analysis.
My goal is to run multiple very similar models after each other and safe only the most relevant results in an array.

The relevant results for me are:
- Steel-Design: max. utilized member
- Stress-Strain-Analysis: max. Mises stress utilization

I am already successfully extracting the max. force of a specific member for each file. I was wandering if I could get the desired results in a similar way.

```ruby
    momentTable = ResultTables.MembersInternalForces(CaseObjectType.E_OBJECT_TYPE_DESIGN_SITUATION, 1, 4)
    maxFor_n = GetMaxValue(momentTable, 'internal_force_n') / 1000
    maxForce_n.append(round(maxFor_n, 3))

```

**Comment by heetrojivadiya** [comment]
Hey, 
unfortunately, we do not have any HLF for design addon result tables at the moment but still you can get result tables for Design addon by Webservice. You may use `get_results_for_steel_design_design_ratios_members_by_design_situation()` method of WS core and get max utilized member by checking max design ratio. Here is small example for your reference:

```py
from RFEM.initModel import Model
from RFEM.Results.resultTables import ConvertResultsToListOfDct,GetMaxValue

Model(False, 'Demo1')
result = ConvertResultsToListOfDct(Model.clientModel.service.get_results_for_steel_design_design_ratios_members_by_design_situation())
max = GetMaxValue(result, 'design_ratio')

for item in result:
    if item.get('design_ratio') == max:
        maximum_utilised = item

print('Maximum utilized member : ', maximum_utilised['member'], ', Loading : ', maximum_utilised['loading'], ', Design ratio : ', maximum_utilised['design_ratio'])
```

output:
```
Maximum utilized member :  3.0 , Loading :  CO3 , Design ratio :  1.1296143531799316
```

You may store your required results as list or dictionary. 


![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/108512789/00aa6a99-09d4-4278-ab46-b7b57140e0dc)

**Reply by tobiasneulichedl** [reply]
Thank you! This looks like a good solution to me.
Is there a way I can get also the min/max stresses from the result table of the stress-strain-analysis?
![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/162106705/f24c9273-e46a-475a-bdd7-cbdd87321289)
